### DISCUSSION TITLE
Documentation - Cisco IOS-XE Flow export and 4 byte ASN (AS 23456)

### DISCUSSION THREAD

**Discussion by NoifP** [akvorado/akvorado#312] [Q&A]
I noticed AS 23456 appearing a lot in my flow data and found it was because my router was not exporting 4-byte ASNs and instead substituting 23546 when a 4-byte ASN was found. The fix was configuration on the exporter. Maybe the following config for IOS-XE can help someone else avoid this pitfall.

```cisco
flow record Akvorado
    match ipv4 tos
    match ipv4 protocol
    match ipv4 source address
    match ipv4 destination address
    match transport source-port
    match transport destination-port
    collect routing source as 4-octet
    collect routing destination as 4-octet
    collect routing next-hop address ipv4
    collect transport tcp flags
    collect interface output
    collect interface input
    collect counter bytes
    collect counter packets
    collect timestamp sys-uptime first
    collect timestamp sys-uptime last
!
sampler random1in100
    mode random 1 out-of 100
!
flow exporter AkvoradoExport
    destination 10.0.0.1 vrf monitoring
    source Loopback2
    transport udp 2055
    version 9
    option sampler-table timeout 10
!
flow monitor AkvoradoMonitor
    exporter AkvoradoExport
    cache timeout inactive 10
    cache timeout active 60
    record Akvorado
! 

interface GigabitEthernet0/0/3
 ip flow monitor AkvoradoMonitor sampler random1in100 input
 ip flow monitor AkvoradoMonitor sampler random1in100 output
!
```

Also note with IOS-XE I ran into the same issue mentioned here: [https://github.com/akvorado/akvorado/issues/89](https://github.com/akvorado/akvorado/issues/89) and the solution to that was to set a default sample rate in akvorado.yaml
```yaml
inlet:
  core:
    default-sampling-rate: 100
```

Maybe these can somehow be integrated into the documentation?

**Comment by vincentbernat** [comment]
Yes, please, can you open a pull request for that? Also, if you have a valid support contract, I would appreciate you open a ticket for #89.

**Reply by NoifP** [reply]
I'll get the PR done shortly. Sorry, I am unable to help with TAC.

**Reply by NoifP** [reply]
Done with PR https://github.com/akvorado/akvorado/pull/316
